nibabel/gifti/gifti.py

```diff
@@ -289,7 +311,6 @@ def from_array(klass,
         if meta is None:
             meta = {}
         cda = klass(darray)
-        cda.num_dim = len(darray.shape)
         cda.dims = list(darray.shape)
         if datatype is None:
             cda.datatype = data_type_codes.code[darray.dtype]
```

Comment:
Given that you've reworked `__init__`, this function wouldn't take much work to clean up so that it ends with something like:

    return klass(darray, encoding, endian, coordsys, ordering, meta)